### PR TITLE
Redsys: Route MIT Exemptions to webservice endpoint

### PR DESCRIPTION
**Pull request by curiousepic**
For MIT-exempt transactions, automatically add the DIRECTPAYMENT field
and route them to the webservice endpoint as a trataPeticion for proper
handling.

Also add flag to force routing to the webservice endpoint and renames
some methods and values for clarity and accuracy.

### CODE REVIEW

File: test/remote/gateways/remote_redsys_sha256_test.rb [hunk 1]
```diff
@@ -198,31 +199,31 @@ def test_successful_auth_purchase_with_stored_credentials
   end
 
   def test_successful_3ds2_purchase_with_mit_exemption
-    options = @options.merge(execute_threed: true, terminal: 12)
-    response = @gateway.purchase(@amount, @threeds2_credit_card, options.merge(sca_exemption: 'MIT', sca_exemption_direct_payment_enabled: true))
+    options = @options.merge(execute_threed: true)
```
Comment (esmitperez) [inline test/remote/gateways/remote_redsys_sha256_test.rb:202]: what's behind the removal of terminal 12?

**Review by Carrigan — CHANGES_REQUESTED** [review]
Rubocop has a couple of failures related to the change that we should take care of before merging, but otherwise the changes all look good and make sense to me :+1: All remote tests pass the same as master.

**Comment by curiousepic** [comment]
Thanks, addressed the Rubocop alerts.

**Review by Carrigan — APPROVED** [review]
Thanks for fixing the linting errors! LGTM :+1:

**Comment by esmitperez** [comment]
I was able to run remote tests! I forgot to adjust the AM hash 🙈 . 

I don't recall the exact steps to reproduce fix in Postman, but we can pair on it

**Review by Carrigan — APPROVED** [review]

**Pull request closed without merging**